app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java

```diff
@@ -180,6 +183,7 @@ protected void onStop() {
     @Override
     protected void onPause() {
         if (mIsPresentingImmersive) {
+            uploadImmersiveElapsedTime();
```

Comment:
I'd move this function to `pauseGeckoViewCompositor`. Immersive mode can be stopped in other situations (e.g. via link traversal) apart from `onBack` & `onPause`

Reply:
@MortimerGoro IIUC, `pauseGeckoViewCompositor ` is the function to begin the immersive mode, so I put my start timer there. `onBack ` & `onPause ` are existing the immersive mode, so I measure the time spent and `uploadImmersiveElapsedTime `.

Reply:
Done. Thanks.